File: Day2/Haskell/GameResult.hs
module GameResult where 
import PlayAction

data GameResult = Defeat | Draw | Victory deriving (Eq)

resultPoints :: GameResult -> Int
resultPoints Defeat = 0
resultPoints Draw = 3
resultPoints Victory = 6

resultFromChar :: Char -> GameResult
resultFromChar 'X' = Defeat
resultFromChar 'Y' = Draw 
resultFromChar _ = Victory

calculateResult :: PlayAction -> PlayAction -> GameResult
calculateResult op you 
    | actionBeats op you = Defeat
    | actionBeats you op = Victory
    | otherwise = Draw

allValues :: [GameResult]
allValues = [Defeat, Draw, Victory]

File: Day2/Java/GameResult.java
public enum GameResult {
    DEFEAT('X', 0),
    DRAW('Y', 3),
    VICTORY('Z', 6);

    private final char letter;
    private final int points;

    GameResult(char letter, int points) {
        this.letter = letter;
        this.points = points;
    }

    public static GameResult calculate(PlayAction opponentAction, PlayAction yourAction) {
        if (opponentAction == yourAction)
            return GameResult.DRAW;
        if (opponentAction.beats(yourAction))
            return GameResult.DEFEAT;
        return GameResult.VICTORY;
    }

    public static GameResult fromLetter(char letter) {
        for (GameResult result : values()) {
            if (result.letter == letter)
                return result;
        }
        return null;
    }

    public int getPoints() {
        return this.points;
    }
}
